### PR TITLE
Add golang-ci-lint action

### PR DESCRIPTION
**Pull request by angrycub**
**Description**
Adding a golang-ci-lint action to the repo

### CODE REVIEW

File: .github/workflows/golang-ci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
```
Comment (lgfa29) [inline .github/workflows/golang-ci-lint.yml:7]: We only use `main`.

File: .github/workflows/golang-ci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+          # Optional: if set to true then the all caching functionality will be complete disabled,
+          #           takes precedence over all other caching options.
+          # skip-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
```
Comment (lgfa29) [inline .github/workflows/golang-ci-lint.yml:44]: I think we can remove all these comments? (and also the `Optional` above).

**Review by lgfa29 — APPROVED** [review]
Minor comments, but LGTM

File: .github/workflows/golang-ci-lint.yml (hunk 1 above)
Comment (angrycub) [inline .github/workflows/golang-ci-lint.yml:7]:
```suggestion
```

File: .github/workflows/golang-ci-lint.yml (hunk 2 above)
Comment (angrycub) [inline .github/workflows/golang-ci-lint.yml:44]:
```suggestion
```

File: .github/workflows/golang-ci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
```
Comment (angrycub) [inline .github/workflows/golang-ci-lint.yml:13]:
```suggestion
```

**Pull request merged**